pkg/controllers/clusterresourceplacement/placement_status_test.go

```diff
@@ -2505,3 +2520,3179 @@ func TestBuildFailedResourcePlacements(t *testing.T) {
 		})
 	}
 }
+
+func TestSetPlacementStatus_useNewConditions(t *testing.T) {
```

Comment:
this UT's attack surface is actually very large, it's a little bit hard to make sure it covers all the combinations. I normally build some basic UT first before testing the top function.

Reply:
let me see if i can split.

Reply:
To validate this SetPlacementStatus func, i still need to setup all other objects and this test size itself cannot be reduced. Added few other UTs to cover internal funcs. At least, we can spot the failure funcs easily when there is a regression.